### ISSUE TITLE
Masking support considering pre-padding ? 

### ISSUE THREAD

**Issue by farmeh** [spyysalo/interleave-layer#2]
Could the interleave layer support masking, assuming the two input sequences are 
**pre-padded** with zeros, i.e., padded with zeros from the **beginning**? 

`   we = np.arange(10).repeat(5).reshape((-1, 5))`
`    de = np.arange(10).repeat(4).reshape((-1, 4)) * 10`
`    w_in = Input(shape=(5,))`
`    d_in = Input(shape=(4,))`
`    w_emb = Embedding(we.shape[0], we.shape[1], weights=[we], mask_zero=True)(w_in)`
`   d_emb = Embedding(de.shape[0], de.shape[1], weights=[de], mask_zero=True)(d_in)`
`  lstm_words = LSTM(output_dim=2, activation=None, return_sequences=True)(w_emb)`
`    lstm_depts = LSTM(output_dim=3, activation=None, return_sequences=True)(d_emb)`
`    out = Interleave(name="OUT")([lstm_words, lstm_depts])`
`    model = Model(input=[w_in, d_in], output=out)`
`    model.compile('adam', 'mse')`
`    sentence1_words = np.asanyarray([[0,0,1,2,3]])`
`    sentence1_depts = np.asanyarray([[0,0,1,2]])`
`    print(model.predict([sentence1_words, sentence1_depts]))`

This way, the interleave layer can be used for more applications.

**Comment by spyysalo** [comment]
f417d3dbc012f6500bc58e38c638a5fb202f33ae implements masking with the assumption that the mask of the interleaved data is identical to the mask of the input deps, which should cover your case.
